scripts/debian/session/tests/run-deb-session-tests.sh

```diff
@@ -0,0 +1,217 @@
+log "Testing deb-session-move.sh"
+"$SESSION_SCRIPTS_DIR/deb-session-move.sh" \
+  "$SESSION_DIR" \
+  "/var/lib/coda/new-ledger1.tar.gz" \
+  "/var/lib/coda/moved-ledger.tar.gz"
+
+assert_exists "$SESSION_DIR/data/var/lib/coda/moved-ledger.tar.gz"
+assert_not_exists "$SESSION_DIR/data/var/lib/coda/new-ledger1.tar.gz"
+
+# ------------------------------------------------------------------------------
+# Test replacing files matching a glob pattern
+# ------------------------------------------------------------------------------
+log "Testing deb-session-replace.sh"
+REPLACEMENT_FILE="$WORKDIR/replacement.json"
+echo 'replaced' > "$REPLACEMENT_FILE"
+"$SESSION_SCRIPTS_DIR/deb-session-replace.sh" \
+  "$SESSION_DIR" \
+  "/var/lib/coda/config_*.json" \
+  "$REPLACEMENT_FILE"
+
+assert_file_equals "$SESSION_DIR/data/var/lib/coda/config_1.json" 'replaced'
+assert_file_equals "$SESSION_DIR/data/var/lib/coda/config_2.json" 'replaced'
+
+# ------------------------------------------------------------------------------
+# Test removing files matching a glob pattern
+# ------------------------------------------------------------------------------
+log "Testing deb-session-remove.sh"
+"$SESSION_SCRIPTS_DIR/deb-session-remove.sh" \
+  "$SESSION_DIR" \
+  "/var/lib/coda/*.tar.gz"
+
+assert_not_exists "$SESSION_DIR/data/var/lib/coda/moved-ledger.tar.gz"
+assert_not_exists "$SESSION_DIR/data/var/lib/coda/new-ledger2.tar.gz"
+
+# ------------------------------------------------------------------------------
+# Test renaming the package
+# ------------------------------------------------------------------------------
+log "Testing deb-session-rename-package.sh"
+"$SESSION_SCRIPTS_DIR/deb-session-rename-package.sh" \
+  "$SESSION_DIR" \
+  "mina-devnet-hardfork"
+
+# Verify the control file was updated
+grep -q '^Package: mina-devnet-hardfork$' "$SESSION_DIR/control/control" \
+  || fail "control file not updated with new package name"
+
+# ------------------------------------------------------------------------------
+# Test saving the session back to a .deb and verifying contents
+# ------------------------------------------------------------------------------
+log "Testing deb-session-save.sh"
+OUTPUT_DEB="$WORKDIR/output.deb"
+"$SESSION_SCRIPTS_DIR/deb-session-save.sh" "$SESSION_DIR" "$OUTPUT_DEB" --verify
+
+NEW_PACKAGE_NAME="$(dpkg-deb --field "$OUTPUT_DEB" Package)"
+if [[ "$NEW_PACKAGE_NAME" != "mina-devnet-hardfork" ]]; then
+  fail "Saved package has unexpected name: $NEW_PACKAGE_NAME"
+fi
+
+# Extract the resulting .deb and verify its contents
+EXTRACT_DIR="$WORKDIR/extracted"
+mkdir -p "$EXTRACT_DIR"
+dpkg-deb -x "$OUTPUT_DEB" "$EXTRACT_DIR" >/dev/null
+
+assert_file_equals "$EXTRACT_DIR/var/lib/coda/config_1.json" 'replaced'
+assert_file_equals "$EXTRACT_DIR/etc/mina/override.json" '{"inserted":true}'
+assert_not_exists "$EXTRACT_DIR/var/lib/coda/moved-ledger.tar.gz"
+
+log "All deb-session scripts executed successfully"
```

Comment:
Thanks for this test, it helps me to understand the usage of the utility scripts.